data_structures/queue/fifo_animal_shelter.py

```diff
@@ -0,0 +1,60 @@
+from node import Node
+
+
+class AnimalShelter:
+    """Create a FIFO AnimalShelter class"""
+    def __init__(self, iter=[]):
+        self.front = None
+        self.back = None
+        self._size = 0
+
+        for item in iter:
+            self.enqueue(item)
+
+    def __repr__(self):
+        """Returns the animal at the front of the queue."""
+        return f'The animal in the shelter longest is: {self.front.val}'
+
+    def enqueue(self, animal=None):
+        """Insert an animal to the back of the queue"""
+        node = Node(animal)
+
+        if animal != 'dog' and animal != 'cat':
```

Comment:
According to the spec, `.enqueue(animal)` takes Dog and Cat objects. Thus you should implement a Dog class and a Cat class to check that the user is enqueueing animals of the proper type.